File: AndroidStudioProjects/HealLotsMobile/app/src/main/java/com/heallots/mobile/ui/adapters/SpecialistAdapter.java
package com.heallots.mobile.ui.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.heallots.mobile.R;
import com.heallots.mobile.utils.MockData;
import java.util.List;

public class SpecialistAdapter extends RecyclerView.Adapter<SpecialistAdapter.ServiceViewHolder> {
    private List<MockData.Service> services;
    private Context context;
    private OnServiceSelectedListener listener;
    private int selectedPosition = -1;
    
    public interface OnServiceSelectedListener {
        void onServiceSelected(MockData.Service service);
    }
    
    public SpecialistAdapter(Context context, List<MockData.Service> services, OnServiceSelectedListener listener) {
        this.context = context;
        this.services = services;
        this.listener = listener;
    }
    
    @NonNull
    @Override
    public ServiceViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_service, parent, false);
        return new ServiceViewHolder(view);
    }
    
    @Override
    public void onBindViewHolder(@NonNull ServiceViewHolder holder, int position) {
        MockData.Service service = services.get(position);
        holder.bind(service, position == selectedPosition);
        holder.itemView.setOnClickListener(v -> {
            int prevSelected = selectedPosition;
            selectedPosition = position;
            notifyItemChanged(prevSelected);
            notifyItemChanged(selectedPosition);
            if (listener != null) listener.onServiceSelected(service);
        });
    }
    
    @Override
    public int getItemCount() {
        return services != null ? services.size() : 0;
    }
    
    public class ServiceViewHolder extends RecyclerView.ViewHolder {
        private LinearLayout serviceCard;
        private TextView serviceIcon, serviceName, serviceSpecialist, serviceDescription;
        
        public ServiceViewHolder(@NonNull View itemView) {
            super(itemView);
            serviceCard = itemView.findViewById(R.id.serviceCard);
            serviceIcon = itemView.findViewById(R.id.serviceIcon);
            serviceName = itemView.findViewById(R.id.serviceName);
            serviceSpecialist = itemView.findViewById(R.id.serviceSpecialist);
            serviceDescription = itemView.findViewById(R.id.serviceDescription);
        }
        
        public void bind(MockData.Service service, boolean isSelected) {
            if (serviceIcon != null) serviceIcon.setText(service.icon);
            if (serviceName != null) serviceName.setText(service.name);
            if (serviceSpecialist != null) serviceSpecialist.setText("with " + service.specialist);
            if (serviceDescription != null) serviceDescription.setText(service.description);
            
            if (serviceCard != null) {
                int bgColor = isSelected ? android.graphics.Color.parseColor("#FFF3E0") : android.graphics.Color.WHITE;
                serviceCard.setBackgroundColor(bgColor);
            }
        }
    }
}

File: AndroidStudioProjects/HealLotsMobile/app/src/main/java/com/heallots/mobile/ui/adapters/SpecialistAdapter.kt
package com.heallots.mobile.features.appointments.book

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.heallots.mobile.R
import com.heallots.mobile.utils.MockData

class SpecialistAdapter(
    private val context: Context,
    private val services: List<MockData.Service>?,
    private val listener: OnServiceSelectedListener?
) : RecyclerView.Adapter<SpecialistAdapter.ServiceViewHolder>() {
    private var selectedPosition = -1

    interface OnServiceSelectedListener {
        fun onServiceSelected(service: MockData.Service)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServiceViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_service, parent, false)
        return ServiceViewHolder(view)
    }

    override fun onBindViewHolder(holder: ServiceViewHolder, position: Int) {
        val service = services?.get(position) ?: return
        holder.bind(service, position == selectedPosition)
        holder.itemView.setOnClickListener {
            val prevSelected = selectedPosition
            selectedPosition = position
            if (prevSelected >= 0) {
                notifyItemChanged(prevSelected)
            }
            notifyItemChanged(selectedPosition)
            listener?.onServiceSelected(service)
        }
    }

    override fun getItemCount(): Int = services?.size ?: 0

    inner class ServiceViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val serviceCard: LinearLayout? = itemView.findViewById(R.id.serviceCard)
        private val serviceIcon: TextView? = itemView.findViewById(R.id.serviceIcon)
        private val serviceName: TextView? = itemView.findViewById(R.id.serviceName)
        private val serviceSpecialist: TextView? = itemView.findViewById(R.id.serviceSpecialist)
        private val serviceDescription: TextView? = itemView.findViewById(R.id.serviceDescription)

        fun bind(service: MockData.Service, isSelected: Boolean) {
            serviceIcon?.text = service.icon
            serviceName?.text = service.name
            serviceSpecialist?.text = "with ${service.specialist}"
            serviceDescription?.text = service.description

            serviceCard?.setBackgroundColor(
                if (isSelected) Color.parseColor("#FFF3E0") else Color.WHITE
            )
        }
    }
}
